typescript/src/shared/errors/sdkError.ts

```diff
@@ -1,10 +1,23 @@
 export class SDKError extends Error {
   constructor(message: string, error: any) {
-    const details = error.body?.errors
-      ?.map((e: any) => e.detailedErrorMessage || e.message)
-      .filter((item: string) => !!item);
+    let details: string[] = [];
+
+    // Handle commercetools API errors
+    if (error.body?.errors) {
+      details = error.body.errors
+        .map((e: any) => e.detailedErrorMessage || e.message)
+        .filter((item: string) => !!item);
+    }
+
+    // Handle Error objects (like the one from base.functions.ts)
+    if (error instanceof Error && error.message) {
+      details.push(error.message);
+    } else if (error.message) {
+      details.push(error.message);
+    }
+
     const errorMessage =
-      message + (details?.length ? ` ${details.join(', ')}` : '');
+      details.length > 0 ? `${message}: ${details.join(', ')}` : message;
```

Comment:
I can see from the previous code there was an `''` fallback empty string here. I think it will be nice if we can have also a fallback default message in a case where `error.message` is `undefined`.